src/ParseServer.js

```diff
@@ -373,6 +373,7 @@ class ParseServer {
     if (process.env.PARSE_SERVER_ENABLE_EXPERIMENTAL_DIRECT_ACCESS === '1') {
       Parse.CoreManager.setRESTController(ParseServerRESTController(appId, appRouter));
     }
+    this.verifyServerUrl(_ref2.publicServerURL);
```

Comment:
_ref2 ?

Reply:
Also? You probably need to add a delay, at least a process.nextTick as the app is not mounted/ not listening yet on the port.

Not sure if we can get that event from Express, i’ll Have a look!

Reply:
That would be ideal to have it execute after an event!  As for `_ref2`, yeah that's gonna get changed before this can be seriously considered, I can't remember where I pulled that from...

Beyond that is this acceptable to put in **ParseServer.js** or do you think it would be more appropriate elsewhere?

Reply:
That’s an excellent place to put it, makes total sense. About the event, I may have been mistaken, perhaps we should wait for the ‘mount’ event, not sure if we can have access to the server that listens in the end. Best chance may be a check after 2s no?

Reply:
We would have to catch when express was listening, but that object is a bit high up. We could do a timing check but I wish we could just get the event. I'll look around a bit more and see if there's anything else, if not we can always do the delay.

Reply:
I do think we can just listen on the api constructed there, it's later returned, and comes back to be listened under whatever the mount path is, under another express instance. I believe when the first one has `listen` called on it, it will chain down, and we'll get our listening callback as expected; without being too far ahead.

Looks like it's used in **ParseWebSocketServer** as such,
```js
...
wss.on('listening', () => {
    logger.info('Parse LiveQuery Server starts running');
});
```